File: application/app/Enums/Expenses/ExpenseSortOptions.php
<?php

namespace App\Enums\Expenses;

enum ExpenseSortOptions: string
{
    case NextDueDate = 'Next Due Date';
    Case Cost = 'Cost';
    case Category = 'Category';
    case CreatedDate = 'Created Date';
    case EndDate = 'End Date';
    case  UpdatedDate = 'Updated Date';
    case RecurrenceRate = 'Recurrence Rate';
    case StartDate = 'Start Date';
    case Name = 'Name';
    case Active = 'Active';

    public function column(): string {
        return match($this) {
            self::NextDueDate => 'next_due_date',
            self::Cost => 'cost',
            self::Category => 'category',
            self::CreatedDate => 'created_at',
            self::EndDate => 'end_date',
            self::RecurrenceRate => 'recurrence_rate',
            self::StartDate => 'start_date',
            self::Active => 'active',
            self::Name => 'name',
            self::UpdatedDate => 'updated_at',
        };
    }

    public static function fromName(string $name): ?self
    {
        foreach (self::cases() as $case) {
            if ($case->name === $name) {
                return $case;
            }
        }
        return null;
    }
}

File: server/BuilderServices/Expenses/ExpenseTableService/Enums/ExpenseSortOptions.cs
using BuilderRepositories;
using BuilderRepositories.Exceptions;

namespace BuilderServices.Expenses.ExpenseTableService.Enums;

public enum ExpenseSortOption
{
    NextDueDate,
    Cost,
    Category,
    CreatedDate,
    EndDate,
    UpdatedDate,
    RecurrenceRate,
    StartDate,
    Name,
    Active
}

public static class ExpenseSortOptionsHelper
{
    public static string GetColumnName(this ExpenseSortOption option)
    {
        return option switch
        {
            ExpenseSortOption.NextDueDate => "e.next_due_date",
            ExpenseSortOption.Cost => "e.cost",
            ExpenseSortOption.Category => "ec.name",
            ExpenseSortOption.CreatedDate => "e.created_at",
            ExpenseSortOption.EndDate => "e.end_date",
            ExpenseSortOption.UpdatedDate => "e.updated_at",
            ExpenseSortOption.RecurrenceRate => "e.recurrence_rate",
            ExpenseSortOption.StartDate => "e.start_date",
            ExpenseSortOption.Name => "e.name",
            ExpenseSortOption.Active => "e.active",
            _ => throw new GenericException("Invalid sort option")
        };
    }

    public static string GetDisplayName(this ExpenseSortOption option)
    {
        return option switch
        {
            ExpenseSortOption.NextDueDate => "Next Due Date",
            ExpenseSortOption.Cost => "Cost",
            ExpenseSortOption.Category => "Category",
            ExpenseSortOption.CreatedDate => "Created Date",
            ExpenseSortOption.EndDate => "End Date",
            ExpenseSortOption.UpdatedDate => "Updated Date",
            ExpenseSortOption.RecurrenceRate => "Recurrence Rate",
            ExpenseSortOption.StartDate => "Start Date",
            ExpenseSortOption.Name => "Name",
            ExpenseSortOption.Active => "Active",
            _ => throw new GenericException("Invalid sort option")
        };
    }
}